From f3d3b1c7dd29d88a4f869213b3d37778fbdcc210 Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Thu, 16 Oct 2014 22:17:53 +0800
Subject: [PATCH 0678/1221] MLK-9617-16 usb: common: otg: clear a_conn flag
 when leaves b host

This patch is clearing a_conn flag when b device leave b_host state.

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit e511cc08879fa97035900ae8626051e473f30cd9)
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/usb/common/usb-otg-fsm.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index 7aabb4d..72f1280 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -90,6 +90,7 @@ static void otg_leave_state(struct otg_fsm *fsm, enum usb_otg_state old_state)
 			fsm->otg_hnp_reqd = 0;
 			fsm->b_bus_req = 0;
 		}
+		fsm->a_conn = 0;
 		break;
 	case OTG_STATE_A_IDLE:
 		fsm->adp_prb = 0;
-- 
1.7.5.4

